test/tests/functional/pbs_soft_walltime.py

```diff
@@ -687,3 +709,336 @@ def test_preempt_order(self):
+        jid = self.server.submit(J)
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 15}, id=jid)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        self.server.deljob(jid, wait=True)
+
+        J = Job(TEST_USER, attrs={'Resource_List.walltime': 16})
+        jid1 = self.server.submit(J)
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 15}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.deljob(jid1, wait=True)
+
+        try:
+            J = Job(TEST_USER, attrs={'Resource_List.walltime': 10})
+            jid1 = self.server.submit(J)
+        except PtlSubmitError as e:
+            self.assertTrue("illegal resource" in e.msg[0])
+
+    def test_soft_suspension(self):
+        """
+        Test that if job is suspended soft_walltime will not get extended
+        """
+
+        J = Job(TEST_USER, {'Resource_List.walltime': 300})
+        jid = self.server.submit(J)
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 7})
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        self.server.sigjob(jid, signal='suspend')
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid)
+
+        time.sleep(7)
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 7}, id=jid)
+
+        self.server.sigjob(jid, signal='resume')
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        time.sleep(5)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {'estimated.soft_walltime':
+                                 (MATCH_RE, '00:00:14|14')}, id=jid)
+
+    def test_soft_less_cput(self):
+        """
+        Test that soft_walltime has no impact on cput enforcement limit
+        """
+
+        script = []
+        script += ['i=0;\n']
+        script += ['while [ $i -ne 0 ] || sleep 0.125;\n']
+        script += ['do i=$(((i+1) % 10000 ));\n']
+        script += ['done\n']
+        j1 = Job(TEST_USER, {'Resource_List.cput': 5})
+        j1.create_script(body=script)
+        jid = self.server.submit(j1)
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 300})
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        # verify that job is deleted when cput limit is reached
+        time.sleep(10)
+        self.server.expect(JOB, 'queue', op=UNSET, id=jid)
+        self.server.log_match(
+            "%s.* resources_used.cput=00:00:05" % jid, regexp=True)
+
+    def test_soft_walltime_resv(self):
+        """
+        Submit a job with soft walltime inside a reservation
+        """
+
+        now = int(time.time())
+
+        a = {'Resource_List.ncpus': 1, 'reserve_start': now + 5,
+             'reserve_end': now + 10}
+        R = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(R)
+        self.scheduler.log_match(
+            rid + ';Reservation Confirmed', max_attempts=5)
```

Comment:
:) This is strange since in your tests you have used the same. I followed the same to keep the tests consistent. I have no objection in changing all of the occurrences if you really think it is important.

Reply:
That was silly of me.  I usually use expect() because that's what is what is commonly used to determine if a reservation was confirmed.  Go ahead and convert them all into expect() calls.